### PR TITLE
Add sorting to file list by tag paragraph

### PR DESCRIPTION
**Pull request by jordan-caldwell**
## Summary
Added sorting to file list by tag by adding a weight field to files.

## Metadata

| Question | Answer |
|----------|--------|
| Did you use a meaningful pull request title? | yes
| Did you apply meaningful labels to the pull request? | yes
| Did you perform a self review first? | yes
| `CHANGELOG` reflects changes? | yes
| Did you perform browser testing? | yes
| Risk level | Low
| Relevant links | https://thinkoomph.jira.com/browse/RIGA-132

### CODE REVIEW

File: ecms_base/themes/custom/ecms/ecms.theme [hunk 1]
```diff
@@ -523,6 +523,7 @@ function ecms_preprocess_paragraph__file_list_by_tag(array &$variables): void {
   $query->condition('bundle', 'file')
     ->condition('field_file_tags', $term_id)
     ->condition('status', 1)
+    ->sort('field_file_list_weight', "ASC")
```
Comment (ben-hamelin) [inline ecms_base/themes/custom/ecms/ecms.theme:526]: What happens to existing entities that don't have a value for this yet - does the query still work?

**Review by ben-hamelin — COMMENTED** [review]
Checking on the sort function call for existing entities, otherwise this looks good

File: ecms_base/themes/custom/ecms/ecms.theme (hunk 1 above)
Comment (jordan-caldwell) [reply ecms_base/themes/custom/ecms/ecms.theme:526]: Hmm, I cant really test that at this point. However, we did do this same thing with person nodes that already existed and it worked fine. My guess is they are null in the database and so it treats it if you were calling a sort on items with values as well as null.

Comment (ben-hamelin) [reply ecms_base/themes/custom/ecms/ecms.theme:526]: Ok, we can test once this hits stage. I will setup a site now with a test paragraph.

**Pull request merged**